tiledb/ml/models/base.py

```diff
@@ -9,13 +15,125 @@ class TileDBModel(abc.ABC):
     store machine learning models (Tensorflow, PyTorch, ect) as TileDB arrays.
     """
 
-    def __init__(self, uri: str):
+    def __init__(self, uri: str, ctx=None, namespace=None):
         """
         Base class for saving machine learning models as TileDB arrays
         and loading machine learning models from TileDB arrays.
         :param uri: str
         """
         self.uri = uri
+        self.tiledb_uri = None
+        self.ctx = ctx
+        self.namespace = namespace
+
+        if self.namespace and self.ctx:
+            s3_prefix = self.get_s3_prefix()
+            if s3_prefix is None:
+                raise HTTPError(
+                    403,
+                    "You must set the default s3 prefix path for ML models in {} profile settings".format(
+                        namespace
+                    ),
+                )
+
+            self.uri = "tiledb://{}/{}".format(
+                namespace, os.path.join(s3_prefix, uri)
+            )
+
+            self.tiledb_uri = "tiledb://{}/{}".format(namespace, uri)
+        else:
+            self.uri = uri
+
+    def get_s3_prefix(self):
+        """
+        Get S3 path from the user profile or organization profile
+        :return: s3 path or error
+        """
+        try:
+            profile = tiledb.cloud.client.user_profile()
+
+            if self.namespace == profile.username:
+                if profile.default_s3_path is not None:
+                    return os.path.join(profile.default_s3_path, CLOUD_MODELS)
+            else:
+                organization = tiledb.cloud.client.organization(self.namespace)
+                if organization.default_s3_path is not None:
+                    return os.path.join(organization.default_s3_path, CLOUD_MODELS)
+        except tiledb.cloud.tiledb_cloud_error.TileDBCloudError as e:
+            raise HTTPError(
+                400,
+                "Error fetching user default s3 path for new ML models {}".format(str(e)),
+            )
+
+        return None
+
+    def get_s3_credentials(self):
```

Comment:
Do we have any unit tests for these functions?

Reply:
As above